feature_engine/timeseries/forecasting/lag_features.py

```diff
@@ -0,0 +1,292 @@
+        self.missing_values = missing_values
+        self.drop_original = drop_original
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn parameters.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training dataset.
+
+        y: pandas Series, default=None
+            y is not needed in this imputation. You can pass None or y.
+        """
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find or check for numerical variables
+        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Adds lag features.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The data to transform.
+
+        Returns
+        -------
+        X_new: Pandas dataframe, shape = [n_samples, n_features + n_operations]
+            The dataframe with the original variables plus the new variables.
+        """
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check if 'X' is a dataframe
+        X = _is_dataframe(X)
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_input_matches_training_df(X, self.n_features_in_)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        if isinstance(self.freq, list):
+            df_ls = []
+            for fr in self.freq:
+                tmp = X[self.variables_].shift(
+                    freq=fr,
+                    axis=0,
+                )
+                df_ls.append(tmp)
+            tmp = pd.concat(df_ls, axis=1)
+
+        elif isinstance(self.periods, list):
+            df_ls = []
+            for pr in self.periods:
+                tmp = X[self.variables_].shift(
+                    periods=pr,
+                    axis=0,
+                )
+                df_ls.append(tmp)
+            tmp = pd.concat(df_ls, axis=1)
+
+        else:
+            tmp = X[self.variables_].shift(
+                periods=self.periods,
+                freq=self.freq,
+                axis=0,
+            )
+
+        tmp.columns = self.get_feature_names_out()
+
+        X = X.merge(tmp, left_index=True, right_index=True, how="left")
+
+        if self.drop_original:
+            X = X.drop(self.variables_, axis=1)
+
+        return X
+
+    def get_feature_names_out(self, input_features=None):
```

Comment:
If this method is only meant to be used internally when .predict() is called should this be a private method? Is there a use case where a user would want to run this method?

Reply:
Sklearn has rolled out this method for all feature extractors and it can be accessed through the pipeline. 

It is meant to return all the features of the dataset if input_features=None, or the features that were created from a certain feature or features if input_features is a list.
https://scikit-learn.org/stable/glossary.html#term-get_feature_names_out

(I think our function here does not do that correctly)

We already got 2 issues from 2 different users requesting this functionality. Plus we should roll it out to adhere to sklearn conventions. We have another PR to roll this out in other transformers: #298